grails-app/services/org/pih/warehouse/data/DataService.groovy

```diff
@@ -382,6 +386,17 @@ class DataService {
         return product
     }
 
+    def findOrCreateProductType(String productTypeName) {
+        def productType = ProductType.findByName(productTypeName)
```

Comment:
I think we should use code for lookup whenever it is available.

Reply:
Code is nullable. Name is unique and cannot be blank. So far there was looking for product type by name (for example in the importing)

Reply:
See my comment above.